JacksonDunstanNativeCollections/NativeElement.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+
+/// <summary>
+/// NativeElement is a wrapper for a single blittable element to be passed between managed and unmanaged code in jobs
+/// </summary>
+/// <typeparam name="T">The blittable type to be stored</typeparam>
+[StructLayout(LayoutKind.Sequential)]
+[NativeContainer]
+[DebuggerDisplay("{Element}")]
+public unsafe struct NativeElement<T> : IDisposable where T : unmanaged
+{
+    // The pointer to the element
+    [NativeDisableUnsafePtrRestriction]
+    T* m_ptr;
+
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+    // Safety
+    AtomicSafetyHandle m_Safety;
+
+    [NativeSetClassTypeToNullOnSchedule]
+    DisposeSentinel m_DisposeSentinel;
+#endif
+
+    // The allocator label
+    Allocator m_AllocatorLabel;
+
+    // Constructor which only takes a label and then calls the main constructor with an instance of T
+    public NativeElement(Allocator label) : this(new T(), label) { }
+
+    // Main constructor logic that takes an element and a label
+    public NativeElement(T element, Allocator label)
+    {
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+        // Label check
+        if (label <= Allocator.None)
+            throw new ArgumentException("NativeElement must be allocated using Job, TempJob or Persistent");
+#endif
+
+        // Label set
+        m_AllocatorLabel = label;
+
+        // Allocate memory for a single T
+        m_ptr = (T*)UnsafeUtility.Malloc(UnsafeUtility.SizeOf<T>(), 1, label);
+
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+        // Create DisposeSentinel and AtomicSafetyHandle
+        DisposeSentinel.Create(out m_Safety, out m_DisposeSentinel, 0, label);
```

Comment:
Support both Unity 2018.2 and Unity 2018.3 versions of this function as other types do:
```csharp
#if UNITY_2018_3_OR_NEWER
	DisposeSentinel.Create(
		out m_Safety,
		out m_DisposeSentinel,
		0,
		allocator);
#else
	DisposeSentinel.Create(out m_Safety, out m_DisposeSentinel, 0);
#endif
```